src/PKSim.Infrastructure/Services/ImportObservedDataTask.cs

```diff
@@ -43,16 +55,71 @@ public class ImportObservedDataTask : IImportObservedDataTask, IObservedDataConf
          _representationInfoRepository = representationInfoRepository;
          _observedDataTask = observedDataTask;
          _parameterChangeUpdater = parameterChangeUpdater;
+         _dialogCreator = dialogCreator;
+         _container = container;
+         _modelingXmlSerializerRepository = modelingXmlSerializerRepository;
       }
 
       public void AddObservedDataToProject() => AddObservedDataToProjectForCompound(null);
 
+      public void AddObservedDataFromConfigurationToProject(ImporterConfiguration configuration) => AddObservedDataFromConfigurationToProjectForCompound(null, configuration, null);
+
+      public void AddObservedDataFromConfigurationToProject(ImporterConfiguration configuration, string dataRepositoryName) => AddObservedDataFromConfigurationToProjectForCompound(null, configuration, dataRepositoryName);
+
+
       public void AddObservedDataToProjectForCompound(Compound compound)
       {
          _executionContext.Load(compound);
          addObservedData(importConfiguration, compound);
       }
 
+      public void AddObservedDataFromXmlToProjectForCompound(Compound compound)
+      {
+         _executionContext.Load(compound);
+         AddObservedDataFromXml(importConfiguration, compound);
```

Comment:
If you blame the file, you will see that importConfiguration comes from you (commit 5580a618) and it is a valid PKSim configuration for the importer. Yes, it does compile.

Reply:
From the diff only, I do not see where this parameter is defined. I will have to load the code in PKSim to review.
And I really don't care who writes the code. If it doesn't make sense, it should change

Reply:
Well, I did not know you were only looking at the diff only. "importConfiguration" is a method on ImportObservedDataTask class and used in some other places. The method configures the columnInfos and yes, it might be better to rename it to columnInfos. Still, from the diff you will have the same doubt meaning, where is it defined since it looks like a local variable but it is actually a private method. I do not recommend changing more than the name of the method.